File: python/copass-management/src/copass_management/tools/list_apps.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def list_apps(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    q = input.get("q")
    cursor = input.get("cursor")
    limit = input.get("limit")
    return await ctx.client.integrations.catalog(
        ctx.sandbox_id,
        q=q if isinstance(q, str) else None,
        cursor=cursor if isinstance(cursor, str) else None,
        limit=int(limit) if isinstance(limit, int) else None,
    )

File: typescript/packages/management/src/tools/list_apps.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const listApps: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  return ctx.client.integrations.catalog(ctx.sandboxId, {
    q: typeof input.q === 'string' ? input.q : undefined,
    cursor: typeof input.cursor === 'string' ? input.cursor : undefined,
    limit: typeof input.limit === 'number' ? input.limit : undefined,
  });
};
